src/main/java/com/imsweb/datagenerator/utils/DistributionUtils.java

```diff
@@ -31,7 +33,8 @@
     private static Map<String, Distribution<CityFrequencyDto>> _DIST_CITIES = new HashMap<>();
     private static Distribution<String> _DIST_STATE;
     private static Map<String, Distribution<String>> _DIST_SITE_AGE_GROUPS = new HashMap<>();
-
+    private static Map<String, Distribution<FacilityFrequencyDto>> _DIST_FACILITIES = new HashMap<>();
```

Comment:
I see here why you called this FacilityFrequencyDto to mimic how it was done with the CityFrequencyDto. But there is a big difference: the city DTO is completely internal, it is meant to represents what we read from the CSV files.

What you read from the facility/physician files should be an internal DTO, and calling it FacilityFrequencyDto is fine.

But what you expose (the list of facilities in the options for example) should not be an object tied to the format of these internal CSV files. Those should be called FacilityDto; the facility generator should build FacilityDto; the facility rule that builds those facilities should use the available frequency and therefore the FacilityFrequencyDto and it should build a FacilityDto from a FacilityFequencyDto. And maybe now both objects will have the same fields and it will seem silly to have those, but I think from a design point-of-view, it's important to not mix them. I should be able for example to add two new fields in the CSV and therefore to the FacilityFrequencyDto, and use them to compute a new field in the FacilityDto, but those two CSV fields would not be added to the FacilityDto class, only the computed one. Does that make sense?

Reply:
I think I understand. In the previous comment, you talked about using a Physician DTO and a Facility DTO, and I assumed you meant PhysicianFrequencyDto and FacilityFrequencyDto. There are no non-frequency DTO's in the project, so I assumed you were talking about them (just writing them in shorthand). I will add these new classes.

Reply:
Sorry, I wasn't clear. I was talking about new non-frequency classes, but I see that wasn't clear. My bad.